### PR TITLE
chore: Create a data mapper for the 4 main Verax objects

### LINKED ISSUE #259: [TASK] Create a data mapper for the 4 main Verax objects

### PR DESCRIPTION
**Pull request by alainncls**
## What does this PR do?

Add a `DataMapper` for the 4 main Verax objects

### Related ticket

Fixes #259 

### Type of change

- [X] Chore
- [ ] Bug fix
- [ ] New feature
- [ ] Documentation update

## Check list

- [ ] Unit tests for any smart contract change
- [ ] Contracts and functions are documented

### CODE REVIEW

File: sdk/examples/attestation/getAttestationIdCounter.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import VeraxSdk from "../../src/VeraxSdk";
```
Comment (fdemiramon) [inline sdk/examples/attestation/getAttestationIdCounter.ts:1]: this example should be removed

File: sdk/examples/attestation/getVersionNumber.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import VeraxSdk from "../../src/VeraxSdk";
```
Comment (fdemiramon) [inline sdk/examples/attestation/getVersionNumber.ts:1]: this example should be removed

File: sdk/examples/module/find.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import VeraxSdk from "../../src/VeraxSdk";
```
Comment (fdemiramon) [inline sdk/examples/module/find.ts:1]: should be removed (only `findBy` and `findOneBy` exist in read)

File: sdk/examples/module/getModulesNumber.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import VeraxSdk from "../../src/VeraxSdk";
```
Comment (fdemiramon) [inline sdk/examples/module/getModulesNumber.ts:1]: should be removed

File: sdk/examples/portal/find.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import VeraxSdk from "../../src/VeraxSdk";
```
Comment (fdemiramon) [inline sdk/examples/portal/find.ts:1]: should be removed

File: sdk/examples/portal/getPortalsCount.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import VeraxSdk from "../../src/VeraxSdk";
```
Comment (fdemiramon) [inline sdk/examples/portal/getPortalsCount.ts:1]: should be removed

File: sdk/examples/schema/find.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import VeraxSdk from "../../src/VeraxSdk";
```
Comment (fdemiramon) [inline sdk/examples/schema/find.ts:1]: should be removed

File: sdk/examples/schema/getSchemasNumber.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
+import VeraxSdk from "../../src/VeraxSdk";
```
Comment (fdemiramon) [inline sdk/examples/schema/getSchemasNumber.ts:1]: should be removed

File: sdk/examples/attestation/findBy.ts [hunk 9]
```diff
@@ -0,0 +1,5 @@
+import VeraxSdk from "../../src/VeraxSdk";
+
+const veraxSdk = new VeraxSdk(VeraxSdk.DEFAULT_LINEA_TESTNET);
+
+console.log(await veraxSdk.attestation.findBy());
```
Comment (fdemiramon) [inline sdk/examples/attestation/findBy.ts:5]: * arguments missing
* implementation missing
* service is `attestationDataMapper` and not `attestation`

File: sdk/examples/portal/findBy.ts [hunk 10]
```diff
@@ -0,0 +1,5 @@
+import VeraxSdk from "../../src/VeraxSdk";
+
+const veraxSdk = new VeraxSdk(VeraxSdk.DEFAULT_LINEA_TESTNET);
+
+console.log(await veraxSdk.portal.findBy());
```
Comment (fdemiramon) [inline sdk/examples/portal/findBy.ts:5]: same

**Review by fdemiramon — CHANGES_REQUESTED** [review]
- + missing examples and implem for write methods

File: sdk/examples/attestation/getAttestationIdCounter.ts (hunk 1 above)
Comment (alainncls) [reply sdk/examples/attestation/getAttestationIdCounter.ts:1]: It feels important to demonstrate web3 calls too, no?

File: sdk/examples/attestation/getVersionNumber.ts (hunk 2 above)
Comment (alainncls) [reply sdk/examples/attestation/getVersionNumber.ts:1]: It feels important to demonstrate web3 calls too, no?

File: sdk/examples/module/getModulesNumber.ts (hunk 4 above)
Comment (alainncls) [reply sdk/examples/module/getModulesNumber.ts:1]: It feels important to demonstrate web3 calls too, no?

File: sdk/examples/portal/getPortalsCount.ts (hunk 6 above)
Comment (alainncls) [reply sdk/examples/portal/getPortalsCount.ts:1]: It feels important to demonstrate web3 calls too, no?

File: sdk/examples/schema/getSchemasNumber.ts (hunk 8 above)
Comment (alainncls) [reply sdk/examples/schema/getSchemasNumber.ts:1]: It feels important to demonstrate web3 calls too, no?

File: sdk/src/dataMapper/ModuleDataMapper.ts [hunk 11]
```diff
@@ -1,3 +1,20 @@
...
+export default class ModuleDataMapper extends BaseDataMapper {
+  typeName = "module";
+  gqlInterface = `{
+        id
+        moduleAddress
+        name
+        description
+  }`;
+
+  async getModulesNumber() {
```
[the remark this answers is not in the document]
Comment (alainncls) [reply sdk/src/dataMapper/ModuleDataMapper.ts:13]: Isn't it a good way to start calling the `ModuleRegistry` contract?

File: sdk/examples/attestation/findBy.ts (hunk 9 above)
Comment (alainncls) [reply sdk/examples/attestation/findBy.ts:5]: * Arguments for this call will be covered in #172 
* `findBy` is implemented and working, even it works like a `findAll` at the moment
* The service was renamed to `attestation` for now, I felt like `attestationDataMapper` is quite long/opinionated for a dev, wdyt?

**Comment by alainncls** [comment]
> Plus missing examples and implem for write methods

Why not, but it feels like this PR will cover everything at once, no?

File: sdk/examples/portal/find.ts (hunk 5 above)
Comment (alainncls) [reply sdk/examples/portal/find.ts:1]: Don't we want an example to find a `Portal`?

File: sdk/examples/module/find.ts (hunk 3 above)
Comment (alainncls) [reply sdk/examples/module/find.ts:1]: Don't we want an example to find a `Module`?

File: sdk/examples/schema/find.ts (hunk 7 above)
Comment (alainncls) [reply sdk/examples/schema/find.ts:1]: Don't we want an example to find a `Schema`?

File: sdk/examples/portal/findBy.ts (hunk 10 above)
Comment (alainncls) [reply sdk/examples/portal/findBy.ts:5]: Don't we want an example to find all `Portals`?

File: sdk/examples/attestation/getAttestationIdCounter.ts (hunk 1 above)
Comment (alainncls) [reply sdk/examples/attestation/getAttestationIdCounter.ts]: Moved to `UtilsDataMapper`

File: sdk/examples/module/find.ts (hunk 3 above)
Comment (alainncls) [reply sdk/examples/module/find.ts]: Moved to `UtilsDataMapper`

File: sdk/examples/portal/find.ts (hunk 5 above)
Comment (alainncls) [reply sdk/examples/portal/find.ts]: Renamed to `findOneById``

File: sdk/examples/portal/getPortalsCount.ts (hunk 6 above)
Comment (alainncls) [reply sdk/examples/portal/getPortalsCount.ts]: Moved to `UtilsDataMapper`

File: sdk/examples/schema/getSchemasNumber.ts (hunk 8 above)
Comment (alainncls) [reply sdk/examples/schema/getSchemasNumber.ts]: Moved to `UtilsDataMapper`

File: sdk/examples/schema/find.ts (hunk 7 above)
Comment (alainncls) [reply sdk/examples/schema/find.ts]: Renamed to `findOneById`

File: sdk/src/dataMapper/ModuleDataMapper.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (alainncls) [reply sdk/src/dataMapper/ModuleDataMapper.ts]: Moved to `UtilsDataMapper`

File: sdk/examples/module/getModulesNumber.ts (hunk 4 above)
Comment (alainncls) [reply sdk/examples/module/getModulesNumber.ts]: Moved to `UtilsDataMapper`

File: sdk/examples/attestation/getVersionNumber.ts (hunk 2 above)
Comment (alainncls) [reply sdk/examples/attestation/getVersionNumber.ts]: Moved to `UtilsDataMapper`

**Pull request merged**